staging/src/k8s.io/kubectl/pkg/cmd/delete/delete.go

```diff
@@ -340,6 +347,14 @@ func (o *DeleteOptions) DeleteResult(r *resource.Result) error {
 		}
 		uidMap[resourceLocation] = responseMetadata.GetUID()
 
+		if o.DryRunStrategy == cmdutil.DryRunNone {
+			// invalidating all local cache if dryrun is not enabled.
```

Comment:
Besides dryrun being disabled, to invalidate the local discovery doc, the `kubectl delete` should:
- delete a CRD / APIService object
- succeed with no error

Reply:
Thanks for the review @roycaihw.

This PR invalidates cache only when https://github.com/kubernetes/kubernetes/blob/f5be5052e3d0808abb904aebd3218fe4a5c2dd82/staging/src/k8s.io/kubectl/pkg/cmd/delete/delete.go#L322 is successful. Is this what you meant?

 For `delete a CRD / APIService object` I could not find a elegant way to infer resource is `CRD` or not in here?, It would be great if you know any example about this.